### PR TITLE
Update `buf` to version 1.2.1, introducing protobuf formatting

### PR DESCRIPTION
**Pull request by christophermaier**
`buf` 1.2.1 now has formatting abilities!

### CODE REVIEW

File: docker-compose.check.yml [hunk 1]
```diff
@@ -46,6 +46,35 @@ services:
...
+    volumes:
+      - *read-only-workdir
+
+  # Use buf to rewrite our protobuf files with the proper formatting.
+  buf-format:
+    image: bufbuild/buf:1.2.1
+    command: ["format", "--verbose", "--write"]
+    environment:
+      # buf tries to write to `~/.cache`, which turns out to be
+      # `/.cache` when running as a non-root user (as we need to do to
```
Comment (jgrillo-grapl) [inline docker-compose.check.yml:66]: "non-root user" or "root user"? If "non-root user" then why does it think `$HOME` is `/`?

**Review by jgrillo-grapl — APPROVED** [review]
LGTM

File: docker-compose.check.yml (hunk 1 above)
Comment (christophermaier) [reply docker-compose.check.yml:66]: Well, "it depends". In this particular case, we're entering the container with the user/group ID that we have on our workstations, which generally works out to `1000:1000`. The `bufbuild/buf` container doesn't actually _have_ a user with that ID, so we're effectively an anonymous user. I _think_ (but am having trouble finding supporting evidence now) that in such a scenario, `/` effectively becomes your home directory... but you won't be able to write to it.

Comment (christophermaier) [reply docker-compose.check.yml:66]: I should probably update that comment, because this does sound confusing as it is written.

Comment (christophermaier) [reply docker-compose.check.yml:66]: Well, I guess this is supporting evidence enough:

```
docker run -it --entrypoint=sh --user=1000:1000 bufbuild/buf:1.2.1 
/ $ whoami
whoami: unknown uid 1000
/ $ echo $HOME
/
/ $ 
```

Comment (jgrillo-grapl) [reply docker-compose.check.yml:66]: interesting

**Pull request merged**